ui/src/alerting/components/builder/CheckConditionsCard.tsx

```diff
@@ -13,30 +13,27 @@ import ThresholdConditions from 'src/alerting/components/builder/ThresholdCondit
 import DeadmanConditions from 'src/alerting/components/builder/DeadmanConditions'
 import BuilderCard from 'src/timeMachine/components/builderCard/BuilderCard'
 
-// Actions & Selectors
-import {getActiveTimeMachine} from 'src/timeMachine/selectors'
-
 // Types
-import {Check, AppState} from 'src/types'
+import {CheckType, AppState} from 'src/types'
 
 interface StateProps {
-  check: Partial<Check>
+  checkType: CheckType
 }
 
 type Props = StateProps
 
-const CheckConditionsCard: FC<Props> = ({check}) => {
+const CheckConditionsCard: FC<Props> = ({checkType}) => {
   let cardTitle: string
   let conditionsComponent: JSX.Element
 
-  if (check.type === 'deadman') {
+  if (checkType === 'deadman') {
     cardTitle = 'Deadman'
-    conditionsComponent = <DeadmanConditions check={check} />
+    conditionsComponent = <DeadmanConditions />
   }
 
-  if (check.type === 'threshold') {
+  if (checkType === 'threshold') {
```

Comment:
There isn't now. since I set `activeTab` to `customCheckQuery` if check.type == "custom", but I think it might be open to breaking during a refactor.

Reply:
but the alternative is also not that great? It would be to check alertBuilder.checkType on line 38 of `ui/src/timeMachine/components/TimeMachine.tsx` I can't convince myself that this is significantly better. I will throw an error to honeybadger here.